### ISSUE TITLE
npm self-upgrade removes itself from path, then fails with permissions error.

### ISSUE THREAD

**Issue by gwk** [npm/npm#17588]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

On macOS High Sierra (10.13 beta 2, update1), I have installed node v8.1.3. When I run `npm i -g npm`, I get the following error (presumably due to not running the command under `sudo`).

At this point, npm has been removed from /usr/local/bin, so the installation is ruined and I have to rerun the installer.

```
npm ERR! path ../../../lib/node_modules/npm/man/man5/npm-folders.5
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall symlink
npm ERR! Error: EACCES: permission denied, symlink '../../../lib/node_modules/npm/man/man5/npm-folders.5' -> '/usr/local/share/man/man5/npm-folders.5'
npm ERR!  { Error: EACCES: permission denied, symlink '../../../lib/node_modules/npm/man/man5/npm-folders.5' -> '/usr/local/share/man/man5/npm-folders.5'
npm ERR!   cause: 
npm ERR!    { Error: EACCES: permission denied, symlink '../../../lib/node_modules/npm/man/man5/npm-folders.5' -> '/usr/local/share/man/man5/npm-folders.5'
npm ERR!      errno: -13,
npm ERR!      code: 'EACCES',
npm ERR!      syscall: 'symlink',
npm ERR!      path: '../../../lib/node_modules/npm/man/man5/npm-folders.5',
npm ERR!      dest: '/usr/local/share/man/man5/npm-folders.5' },
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'symlink',
npm ERR!   path: '../../../lib/node_modules/npm/man/man5/npm-folders.5',
npm ERR!   dest: '/usr/local/share/man/man5/npm-folders.5' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/gwk/.npm/_logs/2017-07-03T20_16_19_226Z-debug.log
```

#### How can the CLI team reproduce the problem?

https://gist.github.com/gwk/38492245911d19de08191a8bee2432f8

### supporting information:

 - `npm -v` prints:  5.0.3
 - `node -v` prints: v8.1.3
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?:  macOS High Sierra (10.13 beta 2, update1)

**Comment by zkat** [comment]
Could you use the install script to install the latest npm? I believe this should be fixed with `npm@5.1.0`.

**Comment by gwk** [comment]
I just tried this with npm 5.2.0, upgrading to 5.3.0 and confirm that it no longer deletes itself.
